rmgpy/rmg/model.py

```diff
@@ -768,23 +768,24 @@ def enlarge(self, newObject=None, reactEdge=False, unimolecularReact=None, bimol
         else:
             # We are reacting the edge
 
-            for i in xrange(numOldCoreSpecies):
-                if unimolecularReact[i]:
-                    # Find reactions involving the species that are unimolecular
-                    reactions = list(react(self.core.species[i].copy(deep=True)))
-                    reactions = [self.inflate(reaction) for reaction in reactions]
-                    self.processNewReactions(reactions, self.core.species[i], None)
+            # Select reactive species that can undergo unimolecular reactions:
+            spcTuples = [(self.core.species[i].copy(deep=True),)
+             for i in xrange(numOldCoreSpecies) if (unimolecularReact[i] and self.core.species[i].reactive)]
 
             for i in xrange(numOldCoreSpecies):
-                for j in xrange(i,numOldCoreSpecies):
+                for j in xrange(i, numOldCoreSpecies):
                     # Find reactions involving the species that are bimolecular
                     # This includes a species reacting with itself (if its own concentration is high enough)
-                    
                     if bimolecularReact[i,j]:
-                        reactions = list(react(self.core.species[i].copy(deep=True), [self.core.species[j]]))
-                        # Consider the latest added core species as the 'new' species
-                        reactions = [self.inflate(reaction) for reaction in reactions]
-                        self.processNewReactions(reactions, self.core.species[j], None)
+                        if self.core.species[i].reactive and self.core.species[j].reactive:
+                            spcTuples.append((self.core.species[i].copy(deep=True), self.core.species[j].copy(deep=True)))
+
+            rxns = list(react(*spcTuples))
+            spcs = [self.retrieveNewSpecies(rxn) for rxn in rxns]
+            
+            for rxn, spc in zip(rxns, spcs):
+                rxn = self.inflate(rxn)    
```

Comment:
from reading the steps in method `inflate`, it doesn't seem to take care of the reactions which have `reverse` attribute which is also a `Reaction` type containing reactants and products, they may still be integer?